GAE/src/org/waterforpeople/mapping/app/web/SurveyRestServlet.java

```diff
@@ -393,6 +395,32 @@ private SurveyInstanceDto findSurveyInstance(Long surveyInstanceId) {
         return dtoList;
     }
 
+    /**
+     * lists all questions for a given survey
+     *
+     * @param surveyId
+     * @return
+     */
+    private List<QuestionDto> listSurveyQuestions(Long surveyId) {
+        List<Question> questions = qDao.listQuestionsBySurvey(surveyId); //useless ordering
+        List<QuestionDto> dtoList = new ArrayList<QuestionDto>();
+        if (questions != null) {
+            for (Question q : questions) {
+                QuestionDto dto = new QuestionDto();
+                DtoMarshaller.copyToDto(q, dto);
```

Comment:
As @valllllll2000 mentioned in one comment, we should already replace this with springs `BeanUtils.copyProperties()`. Be sure to convert the `key().Id()` to `keyId()`